java/quickstart/src/test/java/com/example/cloud/bigtable/quickstart/it/QuickstartIT.java

```diff
@@ -0,0 +1,97 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package com.example.cloud.bigtable.quickstart.it;
+
+import static org.junit.Assert.assertTrue;
+
+import com.example.cloud.bigtable.quickstart.Quickstart;
+import com.google.cloud.bigtable.hbase.BigtableConfiguration;
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.util.UUID;
+import org.apache.hadoop.hbase.HColumnDescriptor;
+import org.apache.hadoop.hbase.HTableDescriptor;
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.client.Connection;
+import org.apache.hadoop.hbase.client.Put;
+import org.apache.hadoop.hbase.client.Table;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/**
+ * Cloud Bigtable Quickstart integration tests
+ */
+@RunWith(JUnit4.class)
+@SuppressWarnings("checkstyle:abbreviationaswordinname")
+public class QuickstartIT {
+  
+  private final String instanceId = System.getProperty("bigtable.test.instance");
+  private final String tableId = formatForTest("my-table");
+  private final String columnFamilyName = "my-column-family";
+  private final String columnName = "my-column";
+  private final String data = "my-data";
+
+  private final String projectId = System.getenv("GOOGLE_CLOUD_PROJECT");
+
+  private String formatForTest(String name) {
+    return name + "-" + UUID.randomUUID().toString().substring(0, 20);
+  }
+
+  @Before
+  public void prepare() throws Exception {
+    try (Connection connection = BigtableConfiguration.connect(projectId, instanceId)) {
+      HTableDescriptor descriptor = new HTableDescriptor(TableName.valueOf(tableId));
+      descriptor.addFamily(new HColumnDescriptor(columnFamilyName));
+      connection.getAdmin().createTable(descriptor);
+
+      // Retrieve the table we just created so we can do some reads and writes
+      Table table = connection.getTable(TableName.valueOf(tableId));
+
+      String rowKey = "r1";
+
+      Put put = new Put(Bytes.toBytes(rowKey));
+      put.addColumn(Bytes.toBytes(columnFamilyName), Bytes.toBytes(columnName),
+          Bytes.toBytes(data));
+      table.put(put);
+
+    }
+  }
+
+  @Test
+  public void quickStart() throws Exception {
+    PrintStream stdOut = System.out;
+    ByteArrayOutputStream bout = new ByteArrayOutputStream();
+    PrintStream out = new PrintStream(bout);
+    System.setOut(out);
+    Quickstart.main(projectId, instanceId, tableId);
+    System.setOut(stdOut);
+    assertTrue(bout.toString().contains(data));
+    Quickstart.main(projectId, instanceId, tableId);
```

Comment:
This last `Quickstart.main` seems unnecessary.

Reply:
done